internal/proxy/oauthproxy.go

```diff
@@ -567,40 +565,23 @@ func (p *OAuthProxy) OAuthCallback(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	// We validate the user information, and check that this user has proper authorization
-	// for the resources requested. This can be set via the email address or any groups.
+	// for the resources requested.
 	//
 	// set cookie, or deny
-	if !p.emailValidator(session.Email) {
-		tags = append(tags, "error:invalid_email")
+
+	errors := options.RunValidators(p.Validators, session)
+	if len(errors) != 0 {
+		tags = append(tags, "error:validation_failed")
 		p.StatsdClient.Incr("application_error", tags, 1.0)
 		logger.WithRemoteAddress(remoteAddr).WithUser(session.Email).Info(
-			"permission denied: unauthorized")
-		p.ErrorPage(rw, req, http.StatusForbidden, "Permission Denied", "Invalid Account")
+			fmt.Sprintf("permission denied: unauthorized: %q", errors))
+		// TODO: As is, the error page given to users provides no extra contract around what failed.
```

Comment:
1. *context
2. This seems like an opportunity to refine the error pages that running these validators will show, and make sure they match where wanted. (depending on where in the logic these validators were ran we _might_ want different errors shown to the end user?) At the moment, there's a couple different variations.